llmfoundry/utils/builders.py

```diff
@@ -495,8 +497,15 @@ def _validate_cfg(icl_cfg: DictConfig):
                 icl_cfg.metric_names = [
                     'InContextLearningMultipleChoiceAccuracy'
                 ]
-            elif icl_cfg.icl_task_type == 'question_answering':
-                icl_cfg.metric_names = ['InContextLearningQAAccuracy']
+            elif icl_cfg.icl_task_type == 'generation_task_with_answers' or icl_cfg.icl_task_type == 'question_answering':
+                if icl_cfg.icl_task_type == 'question_answering':
+                    warnings.warn(
+                        VersionedDeprecationWarning(
```

Comment:
@dakinggg lmk what you think of this? Should I not make it versioned and always keep question_answering as a valid key?

Reply:
deprecating seems reasonable to me if you intend the new name and class to be the usage going forward